src/main/java/week_13/day_3/ProductDao.java

```diff
@@ -6,37 +6,68 @@
 
 public class ProductDao {
     public List<Product> getAllProducts() {
-        List<Product> productList = new ArrayList<>();
-        try (Connection connection = DBConnection.getConnection()) {
-            Statement stmt = connection.createStatement();
-            ResultSet rs = stmt.executeQuery("SELECT * FROM Product");
+        List<Product> products = new ArrayList<>();
+
+        try (Connection connection = DBConnection.getConnection();
+             Statement stmt = connection.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT * FROM Product")) {
+
             while (rs.next()) {
                 Product product = new Product(
                         rs.getInt("id"),
                         rs.getString("name"),
                         rs.getString("category"),
                         rs.getDouble("price"),
-                        rs.getInt("StockQuantity"));
-                productList.add(product);
+                        rs.getInt("stockQuantity"));
+                products.add(product);
             }
         } catch (SQLException e) {
             e.printStackTrace();
         }
-        return productList;
+
+        return products;
+    }
+
+    public List<Product> getProductsByCategoryAndPriceRange(String category, double minPrice, double maxPrice) {
+        List<Product> products = new ArrayList<>();
+        String query = "SELECT * FROM Product WHERE category = ? AND price BETWEEN ? AND ?";
+
+        try (Connection connection = DBConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, category);
+            preparedStatement.setDouble(2, minPrice);
+            preparedStatement.setDouble(3, maxPrice);
+
+            try (ResultSet rs = preparedStatement.executeQuery()) {
+                while (rs.next()) {
+                    Product product = new Product(
+                            rs.getInt("id"),
+                            rs.getString("name"),
+                            rs.getString("category"),
+                            rs.getDouble("price"),
+                            rs.getInt("stockQuantity"));
+                    products.add(product);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return products;
     }
 
     public Product getProduct(int id) {
```

Comment:
I realised I screwed up here when I did not implement a proper primary key as ID and used name instead. I thought it was best to point this out as everything works otherwise